RecoLocalFastTime/FTLClusterizer/interface/MTDArrayBuffer.h

```diff
@@ -0,0 +1,143 @@
+#ifndef RecoLocalTracker_FTLClusterizer_MTDArrayBuffer_H
+#define RecoLocalTracker_FTLClusterizer_MTDArrayBuffer_H
+
+//----------------------------------------------------------------------------
+//! \class MTDArrayBuffer
+//! \brief Class to store ADC counts and times during clustering.
+//!
+//----------------------------------------------------------------------------
+
+// We use FTLHitPos which is an inner class of FTLCluster:
+#include "DataFormats/FTLRecHit/interface/FTLCluster.h"
+
+#include <vector>
+#include <iostream>
+
+class MTDArrayBuffer 
+{
+ public:
+  typedef unsigned int uint;
+  
+  inline MTDArrayBuffer( uint rows, uint cols);
+  inline MTDArrayBuffer( ){}
+  
+  inline void setSize( uint rows, uint cols);
+
+  inline float energy( uint row, uint col) const;
+  inline float energy( const FTLCluster::FTLHitPos&) const;
+  inline float time( uint row, uint col) const;
+  inline float time( const FTLCluster::FTLHitPos&) const;
+  inline float time_error( uint row, uint col) const;
+  inline float time_error( const FTLCluster::FTLHitPos&) const;
+
+  inline uint rows() const { return nrows;}
+  inline uint columns() const { return ncols;}
+
+  inline bool inside(uint row, uint col) const;
+
+  inline void clear(uint row, uint col) 
+  {
+    set_energy( row, col, 0.);
+    set_time( row, col, 0.);
+    set_time_error( row, col, 0.);
+  }
+  inline void clear(const FTLCluster::FTLHitPos& pos) 
+  {
+    clear(pos.row(),pos.col());
+  }
+
+  inline void set( uint row, uint col, float energy, float time, float time_error);
+  inline void set( const FTLCluster::FTLHitPos&, float energy, float time, float time_error);
+
+  inline void set_energy( uint row, uint col, float energy);
+  inline void set_energy( const FTLCluster::FTLHitPos&, float energy);
+  inline void add_energy( uint row, uint col, float energy);
+
+  inline void set_time( uint row, uint col, float time);
+  inline void set_time( const FTLCluster::FTLHitPos&, float time);
+
+  inline void set_time_error( uint row, uint col, float time_error);
+  inline void set_time_error( const FTLCluster::FTLHitPos&, float time_error);
+
+  uint size() const { return hitEnergy_vec.size();}
+
+  /// Definition of indexing within the buffer.
+  uint index( uint row, uint col) const {return col*nrows+row;}
```

Comment:
do the rows start from 0 or from 1?
I see examples of row==nrows.
Depending on the answer, either set_size is usually called with an incorrect argument and needs a +1
or the ```index``` implementation is missing a -1

Reply:
answer to self: there are hits with row==0 and col==0.
So, the set_size calls are wrong.
Please provide a fix before we can wrap up with the review.
Thank you.